misc/testerina/modules/testerina-compiler-plugin/src/main/java/org/ballerinalang/testerina/compiler/TesterinaCompilerPluginUtils.java

```diff
@@ -277,4 +289,28 @@ public static PositionalArgumentNode getPositionalArg(String argName) {
         return NodeFactory.createPositionalArgumentNode(
                 NodeFactory.createSimpleNameReferenceNode(NodeFactory.createIdentifierToken(argName)));
     }
+
+    public static void writeCacheMapAsJson(Map map, Path path, String fileName) {
+        if (!Files.exists(path)) {
+            try {
+                Files.createDirectories(path);
+            } catch (IOException e) {
+                throw new CacheGenException("couldn't create cache directories : " + e.toString());
```

Comment:
Do we need special type of exception for this? IMO, we can use ProjectExeption or LauncherException